althea_types/src/websockets.rs

```diff
@@ -0,0 +1,200 @@
+    },
+    /// Information about the router's connection and bandwidth usage
+    ConnectionDetails {
+        id: Identity,
+        /// Infomation about current exit
+        exit_con: Option<ExitConnection>,
+        /// This is a user set bandwidth limit value, it will cap the users download
+        /// and upload to the provided value of their choosing. Denoted in mbps
+        user_bandwidth_limit: Option<usize>,
+        /// Details of both the Client and Relay bandwidth usage over a given period determined
+        /// by the ops_last_seen_usage_hour in OperatorUpdateMessage. When the device's last
+        /// saved usage hour is the same as the ops last seen, we are up to date. Data sent
+        /// through here gets added to a database entry for each device.
+        user_bandwidth_usage: Option<UsageTrackerTransfer>,
+        /// Current client data usage in mbps computed as the last input to the usage tracker
+        /// so an average of around 5-10 seconds
+        client_mbps: Option<u64>,
+        /// Curent relay data usage in mbps, coputed as the last input to the usage tracker
+        /// so an average of around 5-10 seconds
+        relay_mbps: Option<u64>,
+    },
+}
+
+/// Something the operator may want to do to a router under their control
+#[derive(Debug, Clone, Serialize, Deserialize, Hash, Eq, PartialEq)]
+pub enum OperatorAction {
+    /// Resets the Rita dashboard password. This is the password users use to login
+    /// to the router dashboard, which is distinct from the WiFi password. This
+    /// password is also used for ssh login on the LAN. This reset operation does
+    /// not change that password but it will be changed when the dashboard password
+    /// is set again by the user.
+    ResetRouterPassword,
+    /// This resets the WiFi password to the default 'ChangeMe' and restarts the wifi
+    /// subsystem (without restarting the router).
+    ResetWiFiPassword,
+    // Given a vector of wifitoken, apply these changes to the router
+    SetWifi {
+        token: Vec<WifiToken>,
+    },
+    /// This resets the traffic shaper to 'unlimited' speed for all connections. It can
+    /// be useful when the shaper is showing obviously incorrect values for some peer
+    /// usually caused by bad network transients. While the shaper will eventually recover
+    /// this allows a human to do it right away
+    ResetShaper,
+    /// Fully reboots the router, this includes a power cycle not just a restart of the
+    /// routing processes. For x86 machines this action comes with some risk as devices may
+    /// get stuck in the BIOS if not configured properly.
+    Reboot,
+    /// Restart babeld and rita on the router
+    SoftReboot,
+    /// Sends instructions from op tools about the type of update to perform, either a sysupgrade
+    /// or an opkg update, to be removed after all routers >= beta 19 rc9
+    Update {
+        instruction: UpdateTypeLegacy,
```

Comment:
shouldn't we use the latest update type not the legacy here?

Reply:
we do, with UpdateV2- this is left in to maintain ops compatibility since I've just moved this struct over from the other file